application/F3DStarter.cxx

```diff
@@ -858,7 +863,37 @@ class F3DStarter::F3DInternals
       f3d::window& window = this->Engine->getWindow();
       if (this->AppOptions.Resolution.size() == 2)
       {
-        window.setSize(this->AppOptions.Resolution[0], this->AppOptions.Resolution[1]);
+        F3DOptionsTools::OptionsDict appOptions = F3DOptionsTools::DefaultAppOptions;
+        std::vector<int> defaultRes;
+
+        this->ParseOption(appOptions, "resolution", defaultRes);
+
+        // If user did not set window size, scale it base on fontScale and system zoom scale
+        if (defaultRes != this->AppOptions.Resolution)
+        {
+          window.setSize(this->AppOptions.Resolution[0], this->AppOptions.Resolution[1]);
+        }
+        else
+        {
+          double dpiScaleFactor = 1.0;
+#ifdef WIN32
+          // Get main monitor dpi
+          HDC hdc = GetDC(NULL);
+
+          if (hdc)
+          {
+            const int dpi = GetDeviceCaps(hdc, LOGPIXELSY); // Default return 96
```

Comment:
that looks fine to me, I wonder if we should just add an API in window to get the current scale factor. wdyt @Meakk ?